packages/atomic/src/components/search/atomic-results-per-page/atomic-results-per-page.tsx

```diff
@@ -104,14 +109,27 @@ export class AtomicResultsPerPage implements InitializableComponent {
         style="outline-neutral"
         checked={isSelected}
         ariaLabel={text}
-        onChecked={() => this.resultPerPage.set(choice)}
+        onChecked={() => {
+          this.focusOnProperResultDependingOnChoice(choice);
+          this.resultPerPage.set(choice);
+        }}
         class="btn-page focus-visible:bg-neutral-light"
         part={parts.join(' ')}
         text={text}
       ></RadioButton>
     );
   }
 
+  private focusOnProperResultDependingOnChoice(choice: number) {
+    if (choice < this.resultPerPageState.numberOfResults) {
+      this.bindings.store.state.resultList
+        ?.focusOnFirstResultAfterNextSearch()
+        .then(() => this.scrollToTopEvent.emit());
```

Comment:
Okay, but we need to wait for results to be fetched from the API and rendered before we can scroll correctly.

Otherwise the "top" (ie: where we want to scroll to) changes depending on the content of the result list.

What do you propose instead ?